### PR TITLE
Device atlas API

### PR DESCRIPTION
**Pull request by commuterjoy**
/cc @markstephens This is ready to go once I've got the API details

- [x] Fill out API details and tests
- [x] Wire in to the pipeline
- [x] Benchmark

### CODE REVIEW

**Comment by markstephens** [comment]
Endpoint is: device-lookup-dev.dw.ft.com (view in incognito mode as it doesn't seem to like cookies that much)

I need to send you a key for it.

Post data format is like this:
headers = {
	"Accept":"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
	"Accept-Encoding":"gzip, deflate, sdch",
	"Accept-Language":"en-US;en;en-GB",
	"Cache-Control":"max-age=0",
	"Connection":"keep-alive",
	"Host":"",
	"User-Agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/45.0.2454.85 Safari/537.36"
}
api_key = 'xxxxxxxxxxx'

**Comment by commuterjoy** [comment]
ok, this is ready to go as soon as I've got the key.

File: server/transforms/device-atlas.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+	var headers = event.headers();
+
+	var ua = event.pluck('device.user_agent') || headers['user-agent'];
+
+	if (!ua) {
+		Promise.resolve({});
+	}
+
+	// FIXME only run on 10% of queries
```
Comment (commuterjoy) [inline server/transforms/device-atlas.js]: 5% actually :)

File: tests/server/transforms/device-atlas.spec.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		})
+		deviceAtlas(e)
+			.then(res => {
+				expect(res.isBot).to.equal(true);
+				done();
+			})
+			.catch(err);
+	});
+
+	it.skip('Pluck the user-agent reference from the header if not specified in the event\'s body', done => {
```
Comment (commuterjoy) [inline tests/server/transforms/device-atlas.spec.js]: needs new fixtures. will do this later.

**Comment by tom-parker** [comment]
:+1:  looks good to me

**Comment by markstephens** [comment]
The 5% and "Allow 3000ms to run the pipeline tests" - is the api very slow?

**Comment by commuterjoy** [comment]
> "Allow 3000ms to run the pipeline tests"

This just referrers to another bit of the application that I've been too lazy to stub :)

> 5%

This is just a caution. I'll ramp it up to 100% if everything looks good after a day.

**Pull request merged**